src/line_launcher/duration_slider_listener.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{config, config::midi::MidiSlider, midi, Message};
+use std::sync::{mpsc::Receiver, Arc, Mutex};
+use std::thread;
+use wmidi::Channel;
+
+pub fn listen_for_duration_control_changes(
+    midi_messages_receiver: Receiver<Message>,
+    value: Arc<Mutex<f64>>,
+) {
+    thread::spawn(move || {
+        let slider = MidiSlider {
+            channel: Channel::from_index(15).unwrap(),
+            control_change: config::midi::u8_to_control_function(28).unwrap(),
+        };
+
+        for midi_message in midi_messages_receiver.iter() {
+            let midi_messages = vec![midi_message];
+            if let Some(new_value) = control_value_from_midi_messages(&midi_messages, slider) {
+                let mut value = value.lock().unwrap();
+                *value = get_percent(new_value);
+            }
+        }
+    });
+}
+
+fn control_value_from_midi_messages(midi_messages: &[Message], slider: MidiSlider) -> Option<u32> {
+    let control_value = midi::latest_control_value(slider, midi_messages)?;
+    let new_value = midi::interpolate_control_value(0, 100, control_value);
+
+    Some(new_value)
+}
+
+fn get_percent(value: u32) -> f64 {
+    value as f64
```

Comment:
This all seems a little unsure of how it wants to be expressed in terms of range/units, I went with a float percent (ie 0-100)

Reply:
No I agree with you (I think sometimes I choose percent just because the name is more obvious, for the 0-1 I'm never sure whether eg "ratio" or "fraction" are the right descriptive name)

So updated to pass around a 0-1 value and at that point just made the initial interpolation be between `0.0` and `1.0` so then there's no second multiplication-factor step before using it (ie got rid of `get_percent()` here)